apps/vanilla/src/pages/blackjack/game/gameSession.ts

```diff
@@ -0,0 +1,99 @@
+namespace Game {
+
+	/**
+	 * Game session.
+	 */
+	export class GameSession implements ObserverCore.Observer<PlayerScoreUpdatedEventData>,
+	ObserverCore.Observer<UI.PassButtonClickedEventData>,
+	ObserverCore.Observer<Game.PlayerWonEventData> {
+
+		private readonly gameView: UI.GameView;
+
+		private readonly debugInfo: Game.Debug.DebugInfo;
+
+		private readonly players: Player[];
+
+		private readonly dice: Dice;
+
+		private currentPlayerIndex: number;
+
+		private isGameFinished = false;
+
+		public constructor(players: Player[]) {
```

Comment:
Don't forget `readonly`
```suggestion
		public constructor(players: readonly Player[]) {
```